temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java

```diff
@@ -175,7 +175,10 @@ private ManagedChannel prepareChannel() {
     NettyChannelBuilder builder =
         NettyChannelBuilder.forTarget(options.getTarget())
             .defaultLoadBalancingPolicy("round_robin")
-            .maxInboundMessageSize(MAX_INBOUND_MESSAGE_SIZE);
+            .maxInboundMessageSize(MAX_INBOUND_MESSAGE_SIZE)
+            // to accommodate unbounded Server status messages that may be over the default limit of
+            // 8Kb
+            .maxInboundMetadataSize(MAX_INBOUND_MESSAGE_SIZE / 8);
```

Comment:
Why not use a separate constant for this? Not seeing how this is tied to message size.

Reply:
Well, this is the problem. What should be the number? And if we don’t know, how is a separate constant really better than tangling these two values together?
The benefit of connecting them is that it’s actually easier to say how many MAX_INBOUND_MESSAGE_SIZEs do we allocate on the buffers (like 1.12x right now).
But I don’t care, we can chose some random really large size.